examples/debug_lines_ortho/main.rs

```diff
@@ -0,0 +1,230 @@
+
+        let screen_w = screen_dimensions.width();
+        let screen_h = screen_dimensions.height();
+        let y = t * screen_h;
+
+        debug_lines_resource.draw_line(
+            [0.0, y, 1.0].into(),
+            [screen_w, y + 2.0, 1.0].into(),
+            Srgba::new(0.3, 0.3, 1.0, 1.0),
+        );
+    }
+}
+
+struct ExampleState;
+
+impl SimpleState for ExampleState {
+    fn on_start(&mut self, data: StateData<'_, GameData<'_, '_>>) {
+        // Setup debug lines as a resource
+        data.world.add_resource(DebugLines::new());
+        // Configure width of lines. Optional step
+        data.world
+            .add_resource(DebugLinesParams { line_width: 2.0 });
+
+        // Setup debug lines as a component and add lines to render axis&grid
+        let mut debug_lines_component = DebugLinesComponent::with_capacity(100);
+
+        let (screen_w, screen_h) = {
+            let screen_dimensions = data.world.read_resource::<ScreenDimensions>();
+            (screen_dimensions.width(), screen_dimensions.height())
+        };
+
+        (0..(screen_h as u16))
+            .step_by(50)
+            .map(f32::from)
+            .for_each(|y| {
+                debug_lines_component.add_line(
+                    [0.0, y, 1.0].into(),
+                    [screen_w, (y + 2.0), 1.0].into(),
+                    Srgba::new(0.3, 0.3, 0.3, 1.0),
+                );
+            });
+
+        (0..(screen_w as u16))
+            .step_by(50)
+            .map(f32::from)
+            .for_each(|x| {
+                debug_lines_component.add_line(
+                    [x, 0.0, 1.0].into(),
+                    [x, screen_h, 1.0].into(),
+                    Srgba::new(0.3, 0.3, 0.3, 1.0),
+                );
+            });
+
+        debug_lines_component.add_line(
+            [20.0, 20.0, 1.0].into(),
+            [780.0, 580.0, 1.0].into(),
+            Srgba::new(1.0, 0.0, 0.2, 1.0), // Red
+        );
+
+        data.world.register::<DebugLinesComponent>();
+        data.world
+            .create_entity()
+            .with(debug_lines_component)
+            .build();
+
+        // Setup camera
+        let mut local_transform = Transform::default();
+        local_transform.set_translation_xyz(0.0, screen_h, 10.0);
+        let left = 0.0;
+        let right = screen_w;
+        let bottom = 0.0;
+        let top = screen_h;
+        let znear = 0.0;
+        let zfar = 100.0;
+        data.world
+            .create_entity()
+            .with(Camera::from(Projection::orthographic(
+                left, right, bottom, top, znear, zfar,
+            )))
+            .with(local_transform)
+            .build();
+    }
+
+    fn handle_event(
+        &mut self,
+        _: StateData<'_, GameData<'_, '_>>,
+        event: StateEvent,
+    ) -> SimpleTrans {
+        if let StateEvent::Window(event) = event {
+            if is_close_requested(&event) || is_key_down(&event, VirtualKeyCode::Escape) {
```

Comment:
Hm, this other PR does the opposite: https://github.com/amethyst/amethyst/pull/1708
I don't mind removing it, but I'll wait

Reply:
I'm with @Frizi  to slim it down if we can. #1708 i think is a different case since the example captures the mouse, so closing the window is non-trivial. This example doesn't afaik.

Reply:
Besides that though this is awesome! Love it!

Reply:
cool done!~